derive/src/utils.rs

```diff
@@ -134,31 +134,46 @@ fn crate_access() -> syn::Result<Ident> {
 	}
 }
 
-/// Match `#[codec(crate = ...)]` and return the `...`
-fn codec_crate_path_lit(attr: &Attribute) -> Option<Lit> {
+/// This struct matches `crate = ...` where the ellipsis is a `Path`.
```

Comment:
If we're introducing a helper struct implementing `Parse`, would it be possible to abstract parsing the entire attribute `#[codec(crate = ...)]` rather than a part of it: `crate = ...`?

it seems like it's going to be used once for this specific attribute and if we're introducing more abstractions, it'd be worthwhile to abstract parsing the whole top-level attribute while we're at it.

Reply:
Probably possible, but this is actually the recommended style: match on the path, then put more complicated stuff into the parentheses. See the [`Attribute::parse_args` docs](https://docs.rs/syn/1.0.73/syn/struct.Attribute.html#method.parse_args).

Reply:
Okay then, thanks for the link :slightly_smiling_face: